pkg/envoy/xds_server.go

```diff
@@ -805,12 +805,13 @@ func (s *xdsServer) deleteSecret(name string, wg *completion.WaitGroup, callback
 }
 
 // 'l7lb' triggers the upstream mark to embed source pod EndpointID instead of source security ID
-func getListenerFilter(isIngress bool, useOriginalSourceAddr bool, l7lb bool) *envoy_config_listener.ListenerFilter {
+func getListenerFilter(isIngress bool, useOriginalSourceAddr bool, l7lb bool, proxyPort uint16) *envoy_config_listener.ListenerFilter {
 	conf := &cilium.BpfMetadata{
 		IsIngress:                isIngress,
 		UseOriginalSourceAddress: useOriginalSourceAddr,
 		BpfRoot:                  bpf.BPFFSRoot(),
 		IsL7Lb:                   l7lb,
+		ProxyId:                  uint32(proxyPort),
```

Comment:
nit: maybe add a comment why we're setting the proxy port on the field `ProxyId`. This could look like an error at first glance.

we could do the same in the `accessloger_server.go` were we're doing the same in reverse (interpreting the proxyid from the access log entry as proxyPort) when passing it on to the `EndpointUpdater`.